tokio-macros/src/entry.rs

```diff
@@ -380,7 +380,64 @@ fn parse_knobs(mut input: ItemFn, is_test: bool, config: FinalConfig) -> TokenSt
         }
     };
 
-    let body = input.body();
+    let unit_type = Box::new(parse_quote! { () });
+    let never_type = Box::new(parse_quote! { ! });
+
+    let output_type = match &input.sig.output {
+        // For functions with no return value syn doesn't print anything,
+        // but that doesn't work as `Output` for our boxed `Future`, so
+        // default to `()` (the same type as the function output).
+        syn::ReturnType::Default => &unit_type,
+        syn::ReturnType::Type(_, ret_type) if ret_type == &never_type => &unit_type,
+        syn::ReturnType::Type(_, ret_type) => ret_type,
+    };
+
+    enum ContainsLoopVisitor {
```

Comment:
Obviously have to rename this now

Reply:
I'm using this enum to find a possible `never` type, but it might be tough to make this exhaustive in a forward-compatible way. There's `panic!`, `std::process::exit`, `loop`, and `core::panicking::panic_explicit()` that I know of, but there may be others. 

That doesn't include any user-defined functions that return `!`.

I was getting compiler warnings if I output something like:
```rust
let main_body: Result<(), Error> = loop {
    some_fn().await?;
 };
```

So I'm not sure that this approach really makes sense